R/db-hive.R

```diff
@@ -0,0 +1,43 @@
+#' @export
+db_desc.Hive <- function(x) {
+  info <- DBI::dbGetInfo(x)
+
+  host <- if (info$servername == "") "localhost" else info$servername
+  port <- if (info$port == "") "" else paste0(":", port)
+
+  paste0(
+    info$dbms.name, " ", info$db.version,
+    "[", info$username, "@", host, port,
+    "/", info$dbname , "]")
+
+}
+
+#' @export
+sql_translate_env.Hive <- function(con) {
+
+  sql_variant(
+    scalar = dbplyr::sql_translator(
+      .parent = dbplyr::base_scalar,
+      as.numeric = function(x) build_sql("CAST(", x, " AS DOUBLE)"),
```

Comment:
I think it would be useful to build a helper here:

```R
sql_cast <- function(type) {
  function(x) {
   build_sql("CAST(", x, " AS ", type, ")")
  }
}
```

That should live in the same file as `sql_infix()`.